src/type_translator.ts

```diff
@@ -521,7 +521,13 @@ export class TypeTranslator {
         this.warn('class has no symbol');
         return '?';
       }
-      return '!' + this.symbolToString(type.symbol, /* useFqn */ true);
+      const name = this.symbolToString(type.symbol, /* useFqn */ true);
+      if (name === '(Anonymous class)') {
+        // Values that have anonymous class types produce this name, but the type
+        // appears otherwise identical to a named class.
```

Comment:
Care to comment on intent to handle this differently in the future? I.e. produce a type that closure understands vs '?'.